app/code/Inchoo/Sample04/Setup/Patch/Data/InitialCommentData.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Inchoo\Sample04\Setup\Patch\Data;
+
+use Magento\Framework\Math\Random;
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Inchoo\Sample04\ViewModel\News;
+
+class InitialCommentData implements DataPatchInterface
+{
+    /**
+     * @var ModuleDataSetupInterface
+     */
+    protected $moduleDataSetup;
+
+    /**
+     * @var Random
+     */
+    protected $random;
+
+    /**
+     * @var News
+     */
+    protected $newsViewModel;
+
+    /**
+     * InitialNewsData constructor.
+     * @param ModuleDataSetupInterface $moduleDataSetup
+     * @param Random $random
+     */
+    public function __construct(
+        ModuleDataSetupInterface $moduleDataSetup,
+        Random $random,
+        News $newsViewModel
+    ) {
+        $this->moduleDataSetup = $moduleDataSetup;
+        $this->random = $random;
+        $this->newsViewModel = $newsViewModel;
+    }
+
+    /**
+     * @return string[]
+     */
+    public static function getDependencies()
+    {
+        return [InitialNewsData::class];
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getAliases()
+    {
+        return [];
+    }
+
+    /**
+     * @return $this
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function apply()
+    {
+        $newsCollection = $this->newsViewModel->getNewsList(true);
```

Comment:
nemoj koristiti collection i modele u setup skriptama, napravi sve pomoću "raw query", pod raw mislim na ove Zend metode: getConnection()->select() ....

Reply:
Koja je loša strana korištenja modela i collectiona u setup skripti?

Reply:
U većini slučajeva će raditi, a onda neće i biti će veliki problem jer će se najvjerojatnije manifestirati na production serveru. "Raw" query je brži od korištenja collectiona, modela i repositorija.

Imao sam jednom primjer gdje nije radilo, ali ne mogu se sada sjetiti što je točno bilo.